### PR TITLE
redirect to existing event if one was created within 500 seconds

### PR DESCRIPTION
**Pull request by jshawl**

### CODE REVIEW

**Comment by jshawl** [comment]
cc @mattscilipoti

**Comment by RobertAKARobin** [comment]
Might use `created_at` instead?

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -11,6 +12,10 @@ def create
       event_params["date(5i)"].to_i
     )
     @event.date = date
+    if @event.date.to_i - last_event.date.to_i < 500
+      redirect_to last_event
```
Comment (RobertAKARobin) [inline app/controllers/events_controller.rb]: Can also `return redirect_to last_event`, but whatevs.

File: app/controllers/groups_controller.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@ def show
     @assignments = @submissions.collect(&:assignment).uniq.sort_by(&:due_date)
     @attendances = @group.members_attendances.uniq
     @events = @attendances.collect(&:event).uniq.sort_by(&:date).reverse
-    @event_for_today_already_exists = @events.first.date.strftime("%Y%m%d") == DateTime.now.strftime("%Y%m%d")
+    @event_for_today_already_exists = @events[0] ? @events.first.date.strftime("%Y%m%d") == DateTime.now.strftime("%Y%m%d") : false
```
Comment (RobertAKARobin) [inline app/controllers/groups_controller.rb]: Was throwing an error if the group had no events.

Comment (mattscilipoti) [inline app/controllers/groups_controller.rb]: `@event_for_today_already_exists = @events.any? { |event| event.today? }`

Comment (mattscilipoti) [inline app/controllers/groups_controller.rb]: Prefer this queries db directly.  Looking now.

Comment (mattscilipoti) [inline app/controllers/groups_controller.rb]:
```
class Event < AR::Base
  def self.today
    where("date >= ?", Time.zone.now.beginning_of_day)
  end

  def self.today?
     today.any?
  end
end
```
`@event_for_today_already_exists = @events.today?`

**Pull request merged**